crypto/pkcs12/p12_mutl.c

```diff
@@ -178,7 +255,32 @@ static int pkcs12_gen_mac(PKCS12 *p12, const char *pass, int passlen,
 int PKCS12_gen_mac(PKCS12 *p12, const char *pass, int passlen,
                    unsigned char *mac, unsigned int *maclen)
 {
-    return pkcs12_gen_mac(p12, pass, passlen, mac, maclen, NULL);
+    return pkcs12_gen_mac(p12, pass, passlen, mac, maclen, NID_undef, NID_undef, NULL);
+}
+
+static int pkcs12_hmacnid2mdnid(int hmac_nid)
+{
+    int md_nid = NID_undef;
+
+    switch(hmac_nid) {
+        case NID_hmacWithSHA1:                    md_nid = NID_sha1;                  break;
```

Comment:
there's really no already existing method that maps HMACs to the underlying hash functions?